ISSUE-4: First pass on SBF runners plugin system

DiegoPino opened the pull request:
# What is new here?

See #4 

This is for @giancarlobi for now. 

This pull adds some experiments in top of your work so we can test/discuss and find optimals on how to enqueue, process and save (SBF flavor) output of HOCR, etc.

This is a playground, not ready for propagation of strawberries for mass consumption yet!

 What is in here?

- Basic Module structure
- A generic plugin system we need to refine. Right now it covers a few base classes and Interfaces and a general Binary Post processor Plugin. We will need to bring your HOCR processing to that one to make it a fixed one people can use out of the box
- A new Processor Entity (Config Entity) that keeps the values of this, with its own manager, routing, etc
- Some tooling
- Event handler (needs work to be able to be smart and only trigger background processing if the task is large, or trigger a UI Batch if we feel it can handle the load while the user is watching)
- A Webhook Controller that can be triggered directly by an S3 Event from, e.g, minio (will share example on how to setup that to test later this week)
- A Queue Worker for the Webhook (just a stub for now but will be used when doing other queue workers for the rest of the postprocessors)
- A Playground with some cool Ajax for testing JMESPATH using real ADO data. Really not related to this module but i thought i add it here since we will be using JMESPATH to select (Strawberry pick) where data is taken from and its also a good exercise of sharing knowledge on this via the UI.
- A schema
- Links and Permissions handlers
- License
- Readme
- Most likely typos, bugs, and non-sense!

giancarlobi commented:
@DiegoPino https://github.com/esmero/strawberry_runners/blob/0984bfed3d3b72c98570ba5a74a185814ad59419/strawberry_runners.install#L20 studying your code I found this, probably it would be `'weight'  => 'weight', `

DiegoPino commented:
@giancarlobi i addressed your code review and also opened two follow up issues: 
#6 To deal with the simplest case for EXIF, we can distribute work there
#7 advanced. Make Post Processors hierarchical.

I need help using the right wording to reference to things, so please let me know if things make sense there and/or if you have any questions.

I will also remove the `__destruct()` method on the JSON EVENT that removes temporary files, because we don't know how long they will needed until a plugin managed to process one item from the queue. So i will re-code that to mark each file as in use by each QUEUE item pushed. Then each Post Processor Plugin will have to decrease the usage count once ready. Once a file has no more count (means nobody needs it anymore) i can let Drupal cron get rid of it. Thanks a lot

DiegoPino reviewed (commented):
Will start by merging this or we will never move forward!
